L1Trigger/Phase2L1ParticleFlow/src/TauNNId.cc

```diff
@@ -0,0 +1,75 @@
+#include "L1Trigger/Phase2L1ParticleFlow/interface/TauNNId.h"
+#include "FWCore/ParameterSet/interface/FileInPath.h"
+#include <cmath>
+
+TauNNId::TauNNId() { NNvectorVar_.clear(); }
+TauNNId::~TauNNId() {
+  tensorflow::closeSession(session_);
+  delete graphDef_;
+}
+void TauNNId::initialize(const std::string &iInput, const std::string &iWeightFile, int iNParticles) {
+  edm::FileInPath fp(iWeightFile);
+  graphDef_ = tensorflow::loadGraphDef(fp.fullPath());
```

Comment:
depending on the size of the model graph in memory, might consider using the global cache pattern e.g. https://github.com/cms-sw/cmssw/blob/84d185a6d7613403d070392b8f48ee760c7b337f/RecoMET/METPUSubtraction/plugins/DeepMETProducer.cc

Reply:
what is the limit that you are thinking about ?

Reply:
It's certainly worthwhile for O(100MB), maybe even less

Reply:
done